### PR TITLE
Add ToastNotificationDemoApp to Demos

### PR DESCRIPTION
**Pull request by pmpurifoy**
This PR contains the ToastNotificationsDemoApp that showcases all the features currently available for packaged/unpackaged applications.

Also contains some bugfixes:
- Add LRP to ToastNotificationTests during package setup.
- Include c_toastProtocolArgumentString as an activationKind
- Replace RetrieveAppId() -> RetrieveToastAppId in ToastNotificationManager::Setting

### CODE REVIEW

File: WindowsAppRuntime.sln [hunk 1]
```diff
@@ -273,6 +273,14 @@ Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "ToastNotificationsTestApp",
 EndProject
 Project("{C7167F0D-BC9F-4E6E-AFE1-012C56B48DB5}") = "ToastNotificationsTestAppPackage", "test\TestApps\ToastNotificationsTestAppPackage\ToastNotificationsTestAppPackage.wapproj", "{D6A64926-4988-4C64-A5A8-2C14B1388696}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Demos", "Demos", "{2C6DFE11-B59C-4EE2-8669-AECA96FB69A3}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Tests", "Tests", "{55288381-8BF9-4982-94A4-180ADD59166D}"
+EndProject
```
Comment (danielayala94) [inline WindowsAppRuntime.sln:279]: Who these projects are?

File: test/TestApps/ToastNotificationsDemoApp/main.cpp [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    {
+        // Major.Minor version, MinVersion=0 to find any framework package for this major.minor version
+        const UINT32 c_Version_MajorMinor{ 0x00040001 };
+        const PACKAGE_VERSION minVersion{};
+        RETURN_IF_FAILED(MddBootstrapInitialize(c_Version_MajorMinor, nullptr, minVersion));
+    }
+
+    std::wcout << L"--------------------------------" << std::endl;
+    std::wcout << L"- Toast Notifications Demo App -" << std::endl;
+    std::wcout << L"--------------------------------" << std::endl << std::endl;
```
Comment (danielayala94) [inline test/TestApps/ToastNotificationsDemoApp/main.cpp:120]: Why double `std::endl`?

File: test/TestApps/ToastNotificationsDemoApp/main.cpp [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+}
+
+int main()
+{
+    // Retrieve the app scenario.
+    bool isPackaged{ Test::AppModel::IsPackagedProcess() };
+    if (!isPackaged)
+    {
+        // Major.Minor version, MinVersion=0 to find any framework package for this major.minor version
+        const UINT32 c_Version_MajorMinor{ 0x00040001 };
```
Comment (danielayala94) [inline test/TestApps/ToastNotificationsDemoApp/main.cpp:113]: Is this still valid? Should we need to update this in a frequent basis?

File: test/TestApps/ToastNotificationsDemoApp/main.cpp [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    auto toastNotificationManager{ ToastNotificationManager::Default() };
+
+    // Display the current ToastNotificationSetting for the app.
+    std::wcout << L"Printing ToastNotificationSetting for app: " << GetEnumString(toastNotificationManager.Setting()) << std::endl << std::endl;
+
+    ToastActivationInfo activationInfo{ nullptr };
+    // Create toastActivationInfo depending on packaged | unpackaged scenario.
+    if (isPackaged)
+    {
+        std::wcout << L"Calling ToastActivationInfo::CreateFromActivationGuid with nitro activation guid..." << std::endl;
```
Comment (danielayala94) [inline test/TestApps/ToastNotificationsDemoApp/main.cpp:135]: What does "nitro" mean? Can you rephrase the comment?

File: test/TestApps/ToastNotificationsDemoApp/main.cpp [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+            std::wcout << "Key= " << pair.Key().c_str() << " " << "Value= " << pair.Value().c_str() << std::endl;
+        }
+        std::wcout << std::endl;
+    }
+
+    // Setting up foreground handler for ToastActivation
+    std::wcout << L"Registering foreground handler to receive ToastActivationEventArgs..." << std::endl;
+    winrt::event_token token = toastNotificationManager.ToastActivated([](const auto&, winrt::Microsoft::Windows::ToastNotifications::ToastActivatedEventArgs const& toastArgs)
+    {
+        printf("ToastActivation received foreground!\n");
```
Comment (danielayala94) [inline test/TestApps/ToastNotificationsDemoApp/main.cpp:176]: You could use `std::wcout/std::endl` here right?

File: test/TestApps/ToastNotificationsDemoApp/main.cpp (hunk 3 above)
Comment (pmpurifoy) [reply test/TestApps/ToastNotificationsDemoApp/main.cpp:113]: Because this is a test package it will always be the same major/minor version.

File: test/TestApps/ToastNotificationsDemoApp/main.cpp (hunk 2 above)
Comment (pmpurifoy) [reply test/TestApps/ToastNotificationsDemoApp/main.cpp:120]: To skip a line below it.

Comment (loneursid) [reply test/TestApps/ToastNotificationsDemoApp/main.cpp:120]: You could use "\n" to skip a line. endl is more expensive as it also flush the buffers.

File: test/TestApps/ToastNotificationsDemoPackage/Package.appxmanifest [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+		<com:Extension Category="windows.comServer">
+			<com:ComServer>
+				<com:ExeServer Executable="ToastNotificationsDemoApp\ToastNotificationsDemoApp.exe" Arguments="----AppNotificationActivated:" DisplayName="Toast activator">
+					<com:Class Id="FE8C7374-A28F-4CBE-8D28-4288CBDFD431" DisplayName="Toast activator"/>
+				</com:ExeServer>
+			</com:ComServer>
+		</com:Extension>
+		<uap:Extension Category="windows.protocol" EntryPoint="Windows.FullTrustApplication">
+			<uap:Protocol Name="packagedtoast-protocol">
+				<uap:DisplayName>WindowsAppRuntimeTestProtocol</uap:DisplayName>
```
Comment (danielayala94) [inline test/TestApps/ToastNotificationsDemoPackage/Package.appxmanifest:58]: This approach must be documented in the dev docs. Please sync up with @vaheeshta to get it done.

**Review by danielayala94 — APPROVED** [review]

**Comment by pmpurifoy** [comment]
Moving to clean PR. https://github.com/microsoft/WindowsAppSDK/pull/2079
